new_core_concepts/modules/ROOT/pages/typeql/constraining-data.adoc

```diff
@@ -1,3 +1,98 @@
 = Constraining Data
 
-[placeholder]
\ No newline at end of file
+TypeDB allows additional constraints to be defined on the types in the schema.
+These constraints can restrict the cardinality of an interface, or the domain of values of an attribute-type.
+
+== Cardinality constraints
+Cardinality constraints restrict the number of interfaces of a specific type that may be attached to an instance.
+These are specified using the annotation `@card(x..[y])`, where `x` is the minimum number permitted, and `y` is the maximum number. `y` may be omitted to not enforce an upper limit.
+
+E.g. We can restrict a `friendship` relation to relate exactly two `friends`.
+[,typeql]
+----
+friendship relates friend @card(2);
+----
+
+We can also restrict the number of roles of a specific type an instance plays, or the number of attributes of a specific type it owns.
+E.g. A person can be involved in at most one employment, but own any number of emails.
+[,typeql]
+----
+person plays employment:employee @card(0..1);
+person owns email @card(0..);
+----
+
+=== Cardinality constraints on specialisation
+Cardinality constraints declared on an interface applies to all instances of that interface (including specialisations).
+A specialisation of an interface can also "tighten" the constraint.
```

Comment:
I think our mental model is now simpler - we just require that data conforms to ALL the constraints that are directly declared or on all parents declarations